Requires go 1.12 due to commit e792827e

The commit e792827e introduced `strings.ReplaceAll` which itself is introduced just in go 1.12. So this breaks the compatibility promise of the README with go 1.6+ and breaks building this on debian with go 1.11.

Error message:
```
../../prometheus/promu/cmd/release.go:240:34: undefined: strings.ReplaceAll
```

--- a/README.md
+++ b/README.md
@@ -48,7 +48,7 @@ See documentation example [here](doc/examples/prometheus/.promu.yml)
 
 ## Compatibility
 
-* Go 1.6+
+* Go 1.12+
 
 ## More information
 

